src/ashley/machina_extensions/forum_conversation/forms.py

```diff
@@ -0,0 +1,38 @@
+"""
+    Forum conversation forms
+    ========================
+
+    This module defines forms provided by the ``forum_conversation`` application.
+
+"""
+
+from django.urls import reverse
+from machina.apps.forum_conversation.forms import PostForm as MachinaPostForm
+
+
+class PostForm(MachinaPostForm):
+    """ Overload Machina PostForm to send extra variables to the widget editor """
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        # collect active users for this topic
+        active_users = (
+            self.topic.posts.filter(approved=True, poster__is_active=True)
+            .exclude(poster=self.user)
+            .values("poster__id", "poster__username")
+            .distinct()
+            .order_by("poster__username")
+        )
+
+        # assign the list of active users for the current topic
+        self.fields["content"].widget.attrs["mention_users"] = [
+            {
+                "name": user["poster__username"],
+                "link": reverse(
+                    "forum_member:profile", kwargs={"pk": user["poster__id"]}
+                ),
```

Comment:
I'm wondering if this link should be generated here since it's not used in the AshleyEditor during the writing of the post by its author. The link is only used once the post has been rendered to HTML.

The link to the user profile is generated here and injected to the DraftJS component.
Then, if the post contains a mention, the link is stored in the draft.js JSON state that is POSTed to django-machina.
And then, it is stored in database without any validation.

We do not respect the #1 security rule here : **Don't trust user input**.

The Draft.js State can be modified by a user to inject a link to an external website (e.g. for SEO purpose) in a mention object, that will be rendered like a mention. The integrity between the user that we mention and the link that we render is not guaranteed.

An alternative solution to handle that can be to remove the link from the DraftJS state.
And generate the link it in `editor.decorator.mention`.

WDYT @carofun @sampaccoud ?
Do you think it's a problem we should address ?

Reply:
<img src="https://media.giphy.com/media/Kxz4XFFihiInUclAnc/giphy.gif">

Reply:
_since it's not used in the AshleyEditor during the writing of the post by its author. The link is only used once the post has been rendered to HTML._

not to say you're not right, but juste to add a correction, actually the link is also used directly in the Editor,


 once you've added the @toto, you can see this step on the preview number 3 of this PR, we have the link already active here at this specific moment in the editor.

Reply:
As a matter of fact as discussed, there is no point on being able to see this link in the Editor and as you mentioned we could use later on this information in the database thinking it's a reliable information, best thing to do is to build the link as you suggest on our own rendering the post.
I'll make this fix, thank you !

Reply:
> but the link is calculated in python how would you inject something malicious?

It is calculated in python to be injected in javascript into the the DraftJS editor.
During this step, you can do whatever you want with the draft.js state.

Reply:
True! Didn't catch that :fearful:

Reply:
Fixup has been released ! Thank you both

Reply:
Perfect! :1st_place_medal: